### ISSUE TITLE
Flaky-test: MessageChunkingTest.testMaxPendingChunkMessages

### ISSUE THREAD

**Issue by AnonHxy** [apache/pulsar#18570]
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/pulsar/issues) and found nothing similar.

### Example failure

https://github.com/AnonHxy/pulsar/actions/runs/3524785166/jobs/5910889006

### Exception stacktrace

<details>
<summary>Full exception stacktrace</summary>
<pre><code>
    Error:  testMaxPendingChunkMessages(org.apache.pulsar.client.impl.MessageChunkingTest)  Time elapsed: 0.568 s  <<< FAILURE!
  java.lang.AssertionError: did not expect [25] but found [25]
  	at org.testng.Assert.fail(Assert.java:110)
  	at org.testng.Assert.failEquals(Assert.java:1417)
  	at org.testng.Assert.assertNotEqualsImpl(Assert.java:156)
  	at org.testng.Assert.assertNotEquals(Assert.java:2033)
  	at org.testng.Assert.assertNotEquals(Assert.java:2055)
  	at org.apache.pulsar.client.impl.MessageChunkingTest.testMaxPendingChunkMessages(MessageChunkingTest.java:358)
  	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
  	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
  	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
  	at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)
  	at org.testng.internal.invokers.InvokeMethodRunnable.runOne(InvokeMethodRunnable.java:47)
  	at org.testng.internal.invokers.InvokeMethodRunnable.call(InvokeMethodRunnable.java:76)
  	at org.testng.internal.invokers.InvokeMethodRunnable.call(InvokeMethodRunnable.java:11)
  	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
  	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
  	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
  	at java.base/java.lang.Thread.run(Thread.java:833)
</code></pre>
</details>

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

**Comment by RobertIndie** [comment]
I have opened a PR to unblock it first: https://github.com/apache/pulsar/pull/18572 Then I will open a separate PR to fix it.
